lib/pbench/test/functional/server/test_connect.py

```diff
@@ -29,5 +29,5 @@ def test_connect(self, server_client: PbenchServerClient):
 
         # verify all the required openid-connect fields are present
         if "openid" in endpoints:
-            expected = {"server", "client", "realm", "secret"}
-            assert set(endpoints["openid"]) >= expected
+            expected = {"server", "client", "realm"}
+            assert set(endpoints["openid"]) == expected
```

Comment:
What was wrong with the `>=`?  That prevents the test from failing if the Server suddenly offers additional items in `endpoints["openid"]`...which I think is a good thing.  (We want to be able to change the CUT in compatible ways without having to change the tests at the same time -- changing the tests invalidates them as a means of ensuring the continuity of the CUT, so we want the tests to check what we are supposed to deliver without checking too much else.)

Also, it would be nice if the items in line 32 were sorted; it doesn't actually matter, because the `set()` c'tor will produce an ordered set, but it's nicer for maintenance.

Reply:
Well of course the trick there is whether an old client remains "compatible" if the server changes. At some point we need to think about versioning. Technically, this is `/api/v1` and when we make incompatible changes we need to go to `/api/v2`. If we really follow that, then, yeah, any additional data here under `/api/v1` should remain basically compatible and you're right. Without a formal guarantee to that effect, I'm kinda in favor of the `==`. 😆

Reply:
I am not sure why I made this change, I just spent like 30 mins through the code trying to understand what made me change this so I am just going to revert it back.

Reply:
By "this change", I assume you mean the one to line 33; I expect that you want to retain the change made to line 32 (although, as I said, you could sort the items...)....

Reply:
Ah, I didn't sort it, I should have. :face_exhaling:

Reply:
🥔 🥔 🥔 🥔 🥔 
(Small potatoes....)

Reply:
Okay, Now I did.